### DISCUSSION TITLE
DatabaseConnection Clone Behavior

### DISCUSSION THREAD

**Discussion by panviktor** [SeaQL/sea-orm#2091] [Ideas]
### Description
I'm experiencing an issue where cloning the `AppState` in an Actix web application only works when I use `#[derive(Clone)]`. The `AppState` contains a `DatabaseConnection` and a `redis_pool`. I am using SeaORM version `0.12.10` with certain features enabled.

### Issue Details
- **Framework/Libraries**: Actix, SeaORM
- **Version**: SeaORM `0.12.10`
- **Enabled Features**: `with-json`, `macros`, `sqlx-postgres`, `runtime-actix-rustls`, `chrono`, `postgres-array`

I expect to be able to share the AppState across different parts of the Actix application without having to explicitly derive Clone.

But if I change default-features = false in Cargo.toml - all build well. 

### Code Snippet
```rust
#[derive(Clone)]
pub struct AppState {
    pub(crate) database: DatabaseConnection,
    pub(crate) redis_pool: Pool,
}
```

**Comment by AngelOnFira** [comment]
So if the `mock` feature flag is enabled, then `DatabaseConnection` isn't `Clone`. You can see [that code here](https://docs.rs/sea-orm/0.12.14/src/sea_orm/database/db_connection.rs.html#16). I've create a PR to clarify this in the docs: https://github.com/SeaQL/sea-orm/pull/2109

**Reply by theprimebuilder** [reply]
Thank you by removing the mock feature flag it worked !
